sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3259,23 +3259,11 @@ object GpuOverrides extends Logging {
         Seq(ParamCheck("input",
           (TypeSig.commonCudfTypes + TypeSig.DECIMAL_128 +
               TypeSig.NULL +
-              TypeSig.STRUCT.withPsNote(TypeEnum.STRUCT, "Structs containing " +
-              s"float/double will not be supported.") +
-              TypeSig.ARRAY.withPsNote(TypeEnum.ARRAY, "Arrays containing " +
-              s"floats/doubles will not be supported.")).nested(),
+              TypeSig.STRUCT +
+              TypeSig.ARRAY).nested(),
           TypeSig.all))),
       (c, conf, p, r) => new TypedImperativeAggExprMeta[CollectSet](c, conf, p, r) {
 
-        override def tagAggForGpu(): Unit = {
-          // Fall back to CPU if type is nested and contains floats or doubles in the type tree.
-          // Because NaNs in nested types are not supported yet. 
-          // See https://github.com/NVIDIA/spark-rapids/issues/8808
-          if (c.child.dataType != FloatType && c.child.dataType != DoubleType &&
-              isOrContainsFloatingPoint(c.child.dataType)) {
```

Comment:
Oh yes it's a bit confusing. I will add a comment.

In `GpuCollectBase` we have:
```scala
def child: Expression
override def dataType: DataType = ArrayType(child.dataType, containsNull = false)
```

My change wants to make NaNs UNEQUAL only if `child.dataType` is double or float, otherwise ALL_EQUAL.
  
But when we call `CudfCollectSet` or `CudfMergeSets` we need to pass `dataType` to it. So when I check if `datatype` is `Array(Double)` or `Array(Float)`, what I really want to check is if `child.dataType` is float or double.

(I hope I made it clear...)